### DISCUSSION TITLE
Is there an option to use nlohmann-json and disable exception with clang -fno-exceptions?

### DISCUSSION THREAD

**Discussion by zengtm** [nlohmann/json#2250] [Q&A] [answered]
We found this json class library elegant to use and easy to integrate.

Our next project is in an ARM TEE project. We use ARM LLVM 8.0.12 from Qualcomm, which supports C++11.

However, in the ARM TEE environment, we are running at tight real time constraint and cannot afford the cost of exception handling. The TEE OS layer disables with "-fno-excetions" 

Question: what does it take to build nlohmann-json with -fno-exception, and still pass all tests
?

**Comment by nlohmann** [comment, marked as answer]
See https://nlohmann.github.io/json/home/exceptions/#switch-off-exceptions.

**Reply by nlohmann** [reply]
Quotes:

> Exceptions are used widely within the library. They can, however, be switched off with either using the compiler flag `-fno-exceptions` or by defining the symbol `JSON_NOEXCEPTION`. In this case, exceptions are replaced by `abort()` calls. You can further control this behavior by defining `JSON_THROW_USER` (overriding `throw`), `JSON_TRY_USER` (overriding `try`), and `JSON_CATCH_USER` (overriding `catch`).
> 
> Note that `JSON_THROW_USER` should leave the current scope (e.g., by throwing or aborting), as continuing after it may yield undefined behavior.

**Reply by zengtm** [reply]
Great. Thank you.
